### PR TITLE
Refactor admin endpoints

### PR DESCRIPTION
**Pull request by gijskant**
Fixes [TMT-364](https://jira.thehyve.nl/browse/TMT-364).

### CODE REVIEW

**Comment by gijskant** [comment]
Contains some breaking API changes: some administrative endpoints have been moved to `/v2/admin`.

File: open-api/swagger.yaml [hunk 1]
```diff
@@ -2226,6 +2194,106 @@ paths:
...
+        - v2
+        - admin
+      responses:
+        200:
+          schema:
+            $ref: '#/definitions/UpdateStatus'
+          description: |
+            The cache and patient sets have been cleared. The data for subscribed user queries is updated.
+
+  /v2/admin/system/update_status:
```
Comment (ewelinagr) [inline open-api/swagger.yaml]: Is it a correct description? Looks like a copy of `/v2/admin/system/after_data_loading_update`

File: transmart-core-api/src/main/groovy/org/transmartproject/core/config/UpdateStatus.groovy [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+enum CompletionStatus {
+    CREATED,
+    RUNNING,
+    FAILED,
+    COMPLETED
+}
+
+@Canonical
+class UpdateStatus {
```
Comment (ewelinagr) [inline transmart-core-api/src/main/groovy/org/transmartproject/core/config/UpdateStatus.groovy]: @CompileStatic ?

File: transmart-rest-api/grails-app/controllers/org/transmartproject/rest/SystemController.groovy [hunk 3]
```diff
@@ -18,18 +19,40 @@ class SystemController {
...
             throw new AccessDeniedException('Only allowed for administrators.')
         }
-        systemResource.updateAfterDataLoading(authContext.user)
-        response.status = 200
+        respond systemResource.updateAfterDataLoading()
     }
+
+    /**
+     * Checks if an update task is running:
+     * <code>GET /v2/system/update_status</code>
```
Comment (ewelinagr) [inline transmart-rest-api/grails-app/controllers/org/transmartproject/rest/SystemController.groovy]: <code>GET /v2/admin/system/update_status</code>

File: transmart-rest-api/grails-app/controllers/org/transmartproject/rest/SystemController.groovy [hunk 4]
```diff
@@ -18,18 +19,40 @@ class SystemController {
     /**
      * Clears tree node, counts caches, patient sets and bitsets,
      * updates data for subscribed user queries:
-     * <code>/${apiVersion}/system/after_data_loading_update</code>
+     * <code>GET /v2/system/after_data_loading_update</code>
```
Comment (ewelinagr) [inline transmart-rest-api/grails-app/controllers/org/transmartproject/rest/SystemController.groovy]: <code>GET /v2/admin/system/after_data_loading_update</code>

**Pull request merged**